crates/ethrpc/src/http.rs

```diff
@@ -72,19 +72,33 @@ async fn execute_rpc<T: DeserializeOwned>(
 ) -> Result<T, Web3Error> {
     let body = serde_json::to_string(&request)?;
     tracing::trace!(name = %inner.name, %id, %body, "executing request");
-    let mut request = client
+    let mut request_builder = client
         .post(inner.url.clone())
         .header(header::CONTENT_TYPE, "application/json")
         .header("X-RPC-REQUEST-ID", id.to_string())
         .body(body);
-    if let Some(request_id) = observe::request_id::get_task_local_storage() {
-        request = request.header("X-REQUEST-ID", request_id);
+    match request {
+        Request::Single(Call::MethodCall(method)) => {
+            if let Some(request_id) = observe::request_id::get_task_local_storage() {
+                request_builder = request_builder.header("X-REQUEST-ID", request_id);
+            }
+            request_builder = request_builder.header("X-RPC-METHOD", method.method.clone());
+        }
+        Request::Batch(_) => {
+            if let Some(metadata) = observe::request_id::get_task_local_storage() {
```

Comment:
nit: if we are setting a different header anyway and have a different branch for batch requests, maybe we should also use a different task local with a more appropriate type (e.g. a HashMap<request_id, method>)

Reply:
I wanted to avoid introducing global changes just for this task. But since we already have the "setter", sure, let's add it. But not sure about HashMap, probably a Vector of tuples in order to follow this comment: https://github.com/cowprotocol/services/pull/2750#discussion_r1628911325

Reply:
Hmm, after thinking about this more I think due to type checking / architecture issues this will not be as straight forward as it sounds:
* task local storage can't be generic so we need to have 1 global variable per type
* `http` transport needs to read the task local storage and depending on how we set up the `web3` transport configuration it would either have to read the current task local storage or the about to be introduced variable

If I'm not missing a detail why this should actually be easy to implement I'd say give it a shot to make it nicer but don't get hung up on it.

Reply:
Not sure [that's](https://github.com/cowprotocol/services/pull/2750/commits/62670a3f72db2861e61f06932a540707c0b46a13) worth it. Also, [this path](https://github.com/cowprotocol/services/pull/2750/commits/62670a3f72db2861e61f06932a540707c0b46a13#diff-f48c389d126b8afd133990245b459039d7bbdc83c073d5114471ea5f937ecc33R96-R118) looks redundant for the current implementation, but it covers a possibility that the function was called not from the buffered transport.

Reply:
Let's stick with a simple `Option<String>` to not overcomplicate things.

Reply:
Reverted the commit